appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java

```diff
@@ -966,20 +958,17 @@ public final void setStartedState() {
          // ejb container START state.
 
         try {
-
             initializeInterceptorManager();
 
-            for(Object o : invocationInfoMap.values()) {
-                InvocationInfo next = (InvocationInfo) o;
+            for (InvocationInfo next : invocationInfoMap.values()) {
                 setInterceptorChain(next);
             }
-            for(Object o : this.webServiceInvocationInfoMap.values()) {
+            for (Object o : this.webServiceInvocationInfoMap.values()) {
                 InvocationInfo next = (InvocationInfo) o;
                 setInterceptorChain(next);
             }
-
-        } catch(Exception e) {
-            throw new RuntimeException(e);
```

Comment:
Don't use RuntimeException. What is it? Where we are and why? At least ISE is better - we are where we should not be. Why? I don't know. Wouldn't be better EjbException? But it has it's own processing it could change something. Tests required to do more.

Reply:
ISE is suitable exactly for situations where we reach some illegal state, something is broken and there is no prepared reaction.
I don't change anything without searching - in this case, there are only Exception and Throwable catches. Using instanceof for ISE would be a crime (but yes, catching Throwable is a crime too).

All exceptions from here will end up in this ugly catch block:
```
            } catch (Throwable loadException) {
                logger.log(Level.SEVERE, KernelLoggerInfo.lifecycleException, loadException);
                report.failure(logger, "Exception while loading the app", null);
                report.setFailureCause(loadException);
                tracker.actOn(logger);
            } finally {
```